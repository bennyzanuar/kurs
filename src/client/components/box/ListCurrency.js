import React, { Component } from 'react'
import { currencyCountry } from '../../configs'
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import DeleteIcon from '@material-ui/icons/Delete';
import Button from '@material-ui/core/Button';

function formatNumber(num) {
    var parts = num.toString().split(".");
    parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ",");
    return parts.join(".");
}

class ListCurrency extends Component {
    handleRemove(key, amount){
        const { deletes } = this.props
        deletes(key, amount)
    }
    render(){
        const { rates, amount, handleRemove } = this.props
        return(
            <>
            {
                Object.keys(rates).map((key,index) => {
                return  <List 
                            key={index}
                        >
                            <ListItem>
                                <div className={`box-left`}>
                                    <div className={`width50`}>
                                        <Typography variant="h6" component="h6">
                                            {key}
                                        </Typography>
                                    </div>
                                    <div className={`width50 text-right`}>
                                        <Typography component="p">
                                            {formatNumber((rates[key] * amount).toFixed(4))}
                                        </Typography>
                                    </div>
                                    <Typography component="p">
                                        <strong>{currencyCountry[key]}</strong>
                                    </Typography>
                                    <Typography component="p">
                                        {`1 USD = ${key} ${formatNumber((rates[key] * 1).toFixed(4))}`}
                                    </Typography>
                                </div>
                                <ListItemSecondaryAction>
                                    <IconButton aria-label="Delete" onClick={ (e) => {this.handleRemove(key, amount)} }>
                                        <DeleteIcon/>
                                    </IconButton>
                                </ListItemSecondaryAction>
                            </ListItem>
                        </List>
                })
            }
            </>
        )
    }
}

export default ListCurrency