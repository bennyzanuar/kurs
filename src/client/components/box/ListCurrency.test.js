import React from 'react'
import { shallow } from 'enzyme'
import ListCurrency from './ListCurrency'

describe('<ListCurrency />', () => {
    it('should render list with props', () => {
        const ratesDisplay = {
            IDR: '14102.0011',
            EUR: '0.8854',
            GBP: '0.7720',
            SGD: '1.3568'
        }
        const amount = 10
        const component = shallow(<ListCurrency rates={ratesDisplay} amount={amount}  />)
        expect(component).toMatchSnapshot()
    })
})