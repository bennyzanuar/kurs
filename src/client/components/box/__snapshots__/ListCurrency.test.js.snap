// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListCurrency /> should render list with props 1`] = `
<Fragment>
  <WithStyles(List)
    key="0"
  >
    <WithStyles(ListItem)>
      <div
        className="box-left"
      >
        <div
          className="width50"
        >
          <WithStyles(Typography)
            component="h6"
            variant="h6"
          >
            IDR
          </WithStyles(Typography)>
        </div>
        <div
          className="width50 text-right"
        >
          <WithStyles(Typography)
            component="p"
          >
            141,020.0110
          </WithStyles(Typography)>
        </div>
        <WithStyles(Typography)
          component="p"
        >
          <strong>
            Indonesian Rupiah
          </strong>
        </WithStyles(Typography)>
        <WithStyles(Typography)
          component="p"
        >
          1 USD = IDR 14,102.0011
        </WithStyles(Typography)>
      </div>
      <WithStyles(ListItemSecondaryAction)>
        <WithStyles(IconButton)
          aria-label="Delete"
          onClick={[Function]}
        >
          <pure(DeleteIcon) />
        </WithStyles(IconButton)>
      </WithStyles(ListItemSecondaryAction)>
    </WithStyles(ListItem)>
  </WithStyles(List)>
  <WithStyles(List)
    key="1"
  >
    <WithStyles(ListItem)>
      <div
        className="box-left"
      >
        <div
          className="width50"
        >
          <WithStyles(Typography)
            component="h6"
            variant="h6"
          >
            EUR
          </WithStyles(Typography)>
        </div>
        <div
          className="width50 text-right"
        >
          <WithStyles(Typography)
            component="p"
          >
            8.8540
          </WithStyles(Typography)>
        </div>
        <WithStyles(Typography)
          component="p"
        >
          <strong>
            Euro
          </strong>
        </WithStyles(Typography)>
        <WithStyles(Typography)
          component="p"
        >
          1 USD = EUR 0.8854
        </WithStyles(Typography)>
      </div>
      <WithStyles(ListItemSecondaryAction)>
        <WithStyles(IconButton)
          aria-label="Delete"
          onClick={[Function]}
        >
          <pure(DeleteIcon) />
        </WithStyles(IconButton)>
      </WithStyles(ListItemSecondaryAction)>
    </WithStyles(ListItem)>
  </WithStyles(List)>
  <WithStyles(List)
    key="2"
  >
    <WithStyles(ListItem)>
      <div
        className="box-left"
      >
        <div
          className="width50"
        >
          <WithStyles(Typography)
            component="h6"
            variant="h6"
          >
            GBP
          </WithStyles(Typography)>
        </div>
        <div
          className="width50 text-right"
        >
          <WithStyles(Typography)
            component="p"
          >
            7.7200
          </WithStyles(Typography)>
        </div>
        <WithStyles(Typography)
          component="p"
        >
          <strong>
            British Pound Sterling
          </strong>
        </WithStyles(Typography)>
        <WithStyles(Typography)
          component="p"
        >
          1 USD = GBP 0.7720
        </WithStyles(Typography)>
      </div>
      <WithStyles(ListItemSecondaryAction)>
        <WithStyles(IconButton)
          aria-label="Delete"
          onClick={[Function]}
        >
          <pure(DeleteIcon) />
        </WithStyles(IconButton)>
      </WithStyles(ListItemSecondaryAction)>
    </WithStyles(ListItem)>
  </WithStyles(List)>
  <WithStyles(List)
    key="3"
  >
    <WithStyles(ListItem)>
      <div
        className="box-left"
      >
        <div
          className="width50"
        >
          <WithStyles(Typography)
            component="h6"
            variant="h6"
          >
            SGD
          </WithStyles(Typography)>
        </div>
        <div
          className="width50 text-right"
        >
          <WithStyles(Typography)
            component="p"
          >
            13.5680
          </WithStyles(Typography)>
        </div>
        <WithStyles(Typography)
          component="p"
        >
          <strong>
            Singapore Dollar
          </strong>
        </WithStyles(Typography)>
        <WithStyles(Typography)
          component="p"
        >
          1 USD = SGD 1.3568
        </WithStyles(Typography)>
      </div>
      <WithStyles(ListItemSecondaryAction)>
        <WithStyles(IconButton)
          aria-label="Delete"
          onClick={[Function]}
        >
          <pure(DeleteIcon) />
        </WithStyles(IconButton)>
      </WithStyles(ListItemSecondaryAction)>
    </WithStyles(ListItem)>
  </WithStyles(List)>
</Fragment>
`;
